### PR TITLE
Have a trainee be able to apply with basic data

### PR DESCRIPTION
**Pull request by robsdagreat**
PR Description
Have a user being able to apply for becoming a trainee with basic data and then afterwards he/she can fill the additional information( additional attributes)

Description of tasks that were expected to be completed
A user has to be able to apply for becoming a trainee , provide the additional information required for a trainee and then the application is ready to be considered.

Functionality
After logging in navigate to my applicaiton, then click on Become a trainee button, and fill the form. After successfully submitting the application, you will be prompted to fill the rest of the information so that your application can be considered.

How has this been tested?
This was tested using apollo server 

PR Checklist:
- [x] Have a user apply for becoming  a trainee.
- [x] Have a registered trainee provide additional information for the application.
- [x] Improve the process for administrators to evaluate incoming applications.

Track PR
[#156](https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/156)

Screenshots (If appropriate)

Trainee1

  
Trainee2

### CODE REVIEW

**Review by Philimuhire — APPROVED** [review]

**Review by hbapte — APPROVED** [review]

File: src/resolvers/traineeApplicantResolver.ts [hunk 1]
```diff
@@ -105,40 +105,56 @@ export const traineeApplicantResolver: any = {
...
+        await session.abortTransaction();
+        throw error;
+      } finally {
+        session.endSession();
+      }
+    }
 
-      await sendEmailTemplate(emailTest, "Applicant invitation",
-        `Hello ${emailTest.split("@")[0]}`,
-        `You are invited to join Devpulse application ${cycle.name} <br/>
```
Comment (mugishaj092) [inline src/resolvers/traineeApplicantResolver.ts:134]: Why did you delete this sendEmailTemplate function?

**Review by mugishaj092 — CHANGES_REQUESTED** [review]

File: src/resolvers/traineeApplicantResolver.ts (hunk 1 above)
Comment (robsdagreat) [reply src/resolvers/traineeApplicantResolver.ts:134]: The reason I removed it is that this resolver is for creating a trainee and we're sending an email for application while is applying?

**Review by mugishaj092 — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]
